libs/staging-provider/src/Provider/StorageApiWorkspace.php

```diff
@@ -0,0 +1,98 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Keboola\StagingProvider\Provider;
+
+use Keboola\StagingProvider\Exception\StagingProviderException;
+use Throwable;
+
+readonly class StorageApiWorkspace
+{
+    public function __construct(
+        public string $id,
+        public ?string $backendSize,
+        /** @var array{
+         *     host?: string,
+         *     warehouse?: string,
+         *     database?: string,
+         *     schema?: string,
+         *     user?: string,
+         *     password?: string,
+         *     account?: string,
+         *     container?: string,
+         *     connectionString?: string} $connection
+         * */
+        public array $credentials,
+    ) {
+    }
+
+    public static function fromDataArray(array $data): self
```

Comment:
Zajimavy, jak se kopa trid da smrsknout do takovyho malyho mista :slightly_smiling_face: Je to super, ale vidim tu jeden nedostatek - struktura `credentials` je zavisla na typu workspace, ale ten typ tu potom zpetne nevyctes. Jinak receno, pokud ti nekam prijde `StorageApiWorkspace`, musi ti jeste jinou cestou prijit o jaky typ workspace se jedna, aby si vedel, co ocekavat v credentials.

A znas me, ja rad objekty se striktni strukturou :wink: takze me hned napada, kam by se do dalo poslat dal. Nevim jestli jsme se uz nekdy nebavili o tom, ze by ty credentials byly jako class? V podstate by stacily 4 (eventuelne 7 no) readonly data class.

Reply:
Je to tak, taky mě to na tom mrdalo, ale úplně jsem nepřišel na rozumnej způsob jak to udělat. Ona je tam pak zase větev  těch Credentials tríd, jenže ty zase reprezentují něco jinýho. A celý je to vlastně na kokot protože kdyby Storage API klient vracel objekty, tak by nemusely bejt v prdeli rybníky.

Stačilo by ti tam teď přidat type? A pak bysme viděli jestli by byl někde potřeba? (protože zatím nikde potřeba není)

Reply:
dal jsem tam `backend`, což je vlastně velice logický, když je tam backend size, potřeba to zatím na nic není, ale future self když bude psát switch StorageApiWorkspace::backend tak by se měl pokusit to nenapsat

Reply:
> Stačilo by ti tam teď přidat type?

Jo, urcite. To s tema objektama bylo spis takovy zbozny prani :smile: